test/deprecated/update-deprecated-links

```diff
@@ -1,14 +1,21 @@
-#!/bin/bash
+#!/usr/bin/env bash
```

Comment:
I think this could use some explanation/rationale in the commit message. We don't do that for any other bash script of ours, and for example `test/update-test-cmd-list` which is also invoked by pre-commit has a `/bin/bash` one.

Reply:
I have also updated `test/update-test-cmd-list` and `runLint`. I added reasoning in the commit message, but if you think we don't have to change the shebang, I can drop the commit from the PR as I'm using Linux and do not have the issue with the path of Bash.

I kept the shebang of `test/docker/*/install-packages.sh` because these are scripts for Linux, but these might also be updated for consistency.

Reply:
This is fine with me. Could fine tune the message of the latter commit now that it includes more than just `update-deprecated-links`, pre-approved with that change. Thanks!